src/uct/sm/cma/cma_ep.c

```diff
@@ -53,69 +82,40 @@ ucs_status_t uct_cma_ep_common_zcopy(uct_ep_h tl_ep,
                                                      const struct iovec *,
                                                      unsigned long,
                                                      unsigned long),
-                                     char *fn_name)
+                                     const char *fn_name)
 {
-    ssize_t ret;
-    ssize_t delivered = 0;
-    size_t iov_it;
-    size_t iov_it_length;
-    size_t iov_slice_length;
-    size_t iov_slice_delivered;
-    size_t local_iov_it;
-    size_t length = 0;
+    uct_cma_ep_t *ep = ucs_derived_of(tl_ep, uct_cma_ep_t);
+    size_t iov_idx   = 0;
+    ucs_status_t status;
+    size_t local_iov_cnt;
+    size_t length;
+    size_t cur_iov_cnt;
     struct iovec local_iov[UCT_SM_MAX_IOV];
     struct iovec remote_iov;
-    uct_cma_ep_t *ep = ucs_derived_of(tl_ep, uct_cma_ep_t);
 
-    do {
-        iov_it_length = 0;
-        local_iov_it = 0;
-        for (iov_it = 0; iov_it < ucs_min(UCT_SM_MAX_IOV, iovcnt); ++iov_it) {
-            iov_slice_delivered = 0;
-
-            /* Get length of the particular iov element */
-            iov_slice_length = uct_iov_get_length(iov + iov_it);
-
-            /* Skip the iov element if no data */
-            if (!iov_slice_length) {
-                continue;
-            }
-            iov_it_length += iov_slice_length;
-
-            if (iov_it_length <= delivered) {
-                continue; /* Skip the iov element if transferred already */
-            } else {
-                /* Let's assume the iov element buffer can be delivered partially */
-                if ((iov_it_length - delivered) < iov_slice_length) {
-                    iov_slice_delivered = iov_slice_length - (iov_it_length - delivered);
-                }
-            }
-
-            local_iov[local_iov_it].iov_base = (void *)((char *)iov[iov_it].buffer +
-                                                        iov_slice_delivered);
-            local_iov[local_iov_it].iov_len  = iov_slice_length - iov_slice_delivered;
-            ++local_iov_it;
-        }
-        if (!delivered) {
-            length = iov_it_length; /* Keep total length of the iov buffers */
-        }
+    remote_iov.iov_base = UCS_PTR_BYTE_OFFSET(remote_addr, 0);
```

Comment:
is this to cast to `(void*)` ?

Reply:
> is this to cast to `(void*)` ?

yep, I just want to not break @hoopoepg fixes for PGI